### PR TITLE
WIP: CI with travis-ci

### LINKED ISSUE #125: Continuous testing needs to be added
Continuous testing has to be added to the system. Many vendors are present:

- https://travis-ci.org/
- https://circleci.com/

Use some CI system and perform the following:

 - code format check with [black](https://github.com/python/black)
- run tests (with pytest maybe)?

### PR DESCRIPTION
**Pull request by rishabhKalakoti**
I made a custom CI configuration, not sure if this one is correct. I had a few problems.
* `pytest` had to be run in `pipenv`. Therefore I had to add the installation commands (which I added under `scripts`). I think they should not be added under scripts, I found another implementation on the web using `make`, but could not implement it (made a lot of commits during testing so had to fork again, 'coz I got it messed up).
* The present format in which tests are written, we have to run `server.py` first and then `pytest`, I think this can be done better, I found an implementation with https://pypi.org/project/WebTest/ which looks better to me. So I commented the tests and make a `demo` test case with `assert 1==1`
* `travis-ci` would have to be added to PyJaipur/PyJudge by someone with admin access.

Roadmap:
- [x] configure `travis-ci` properly
- [x] formatting check with black in CI
- [x] test the `bottle` framework correctly (*using `WebTest` maybe*)
- [ ] add more tests

This can be done in separate PRs too, because tests might mess things up :)

fixes #125

### CODE REVIEW

**Comment by rishabhKalakoti** [comment]
As of now, its working fine on my repository.
(installation also as tests)

File: test_server.py [hunk 1]
```diff
@@ -3,7 +3,9 @@
 #import server as app
 import requests
 
-
+def test_demo():
+    assert 1==1
+"""
```
Comment (theSage21) [inline test_server.py]: are you trying to comment?

Comment (rishabhKalakoti) [reply test_server.py]: Yes, I commented the present tests, as they are outdated and we also had to run `server.py` for the tests, so I will rewrite the tests after `travis` is configured properly.

Comment (theSage21) [reply test_server.py]: Let's use actual comments instead of multiline strings? Shows up better in a diff too.

Comment (rishabhKalakoti) [reply test_server.py]: So, I will have to comment all lines...

Comment (theSage21) [reply test_server.py]: whatever editor you are using will have support for block commenting. Actually forget this. No point in enforcing this. Anyways this is going to go out right?

Comment (rishabhKalakoti) [reply test_server.py]: Done. Yes it will go when I rewrite the tests.

**Comment by rishabhKalakoti** [comment]
@theSage21 I need help in better configuration settings for `travis`. Also what tests should I add apart from testing that the web server is running well.

**Comment by theSage21** [comment]
hmm. I'm in a fix here.

1. I don't have any experience with travis so help with that is going to be delayed
2. I've usually built API servers which don't render html. we simply do an api level testing in that case. Again I'm lost here.

so ... I'll say let's do 2 things:

1. code formatting checks
2. api level checks
    - (if I asked for a contest is there some question in it?)
    - if I'm not logged in make sure I can't submit

**Comment by rishabhKalakoti** [comment]
How do i check if a user is logged in? I got no idea how to handle cookies (they are part of browser) in testing (i will try googling). One fix I can do (not sure if it will work), is check for a user in database and then send form queries one by one as we do in the browser.

**Comment by rishabhKalakoti** [comment]
(if I asked for a contest is there some question in it?)
Can I check the above directly in the database?

**Comment by rishabhKalakoti** [comment]
@theSage21 let's wrap this one here. I am a little busy and we are also changing the question/contest framework, so API tests can be covered after that. This one would be enough to setup CI with format check and server check. An admin will need to activate this repository for travis-ci to complete this.

**Comment by theSage21** [comment]
@rishabhKalakoti why don't you set up travis for this repo. I've added you as an admin. Once you're done you can de-escalate your privileges to prevent accidental admin actions on the repo.

**Comment by rishabhKalakoti** [comment]
@theSage21 I will get it done on Saturday.

**Pull request merged**